src/adaptors/kava-mint/index.js

```diff
@@ -0,0 +1,131 @@
+const axios = require('axios');
+const superagent = require('superagent');
+const utils = require('../utils');
+const USDX_ID = 'usdx';
+
+const getPrices = async (addresses) => {
+  const prices = (
+    await superagent.post('https://coins.llama.fi/prices').send({
+      coins: addresses,
+    })
+  ).body.coins;
+
+  const pricesObj = Object.entries(prices).reduce(
+    (acc, [address, price]) => ({
+      ...acc,
+      [address.split(':')[1].toLowerCase()]: price.price,
+    }),
+    {}
+  );
+  return pricesObj;
+};
+
+const main = async () => {
+  const parametersUrl = 'https://api2.kava.io/cdp/parameters';
+  const dispoistedUrl = 'https://api2.kava.io/cdp/totalCollateral';
+  const principalUrl = 'https://api2.kava.io/cdp/totalPrincipal';
+
+  const [parametersCall, dispoistedCall, principalCall] = (
+    await Promise.all([
+      axios.get(parametersUrl),
+      axios.get(dispoistedUrl),
+      axios.get(principalUrl),
+    ])
+  ).map((e) => e.data.result);
+
+  const parameters = parametersCall.collateral_params.map((e) => {
+    return {
+      ...convertSymbol(e.denom),
+      stability_fee: e.stability_fee,
+      type: e.type,
+      liquidation_ratio: e.liquidation_ratio,
+      debt_limit: e.debt_limit.amount,
+    };
+  });
+
+  const dispoisted = dispoistedCall.map((e) => {
+    const info = convertSymbol(e.amount.denom);
+    return {
+      ...info,
+      amount: Number(e.amount.amount / 10 ** (info?.decimals || 0)),
+      type: e.collateral_type,
+    };
+  });
+
+  const borrowed = principalCall.map((e) => {
+    const info = convertSymbol(e.collateral_type.split('-')[0]);
+    return {
+      ...info,
+      amount: Number(e.amount.amount / 10 ** (info?.decimals || 0)),
+      type: e.collateral_type,
+    };
+  });
+
+  const coins = dispoisted.map((e) => `coingecko:${e.id}`);
+  const prices = await getPrices([...coins, `coingecko:${USDX_ID}`]);
+
+  return parameters
+    .filter((e) => e.id)
+    .map((pool) => {
+      const parameter = parameters.find((e) => e.type === pool.type);
+      const collateral = dispoisted.find((e) => e.type === pool.type);
+      const _borrowed = borrowed.find((e) => e.type === pool.type);
+      const totalSupplyUsd = collateral.amount * prices[pool.id.toLowerCase()];
+      const totalBorrowUsd = _borrowed.amount * prices[pool.id.toLowerCase()];
+      const ltv = (1 / Number(parameter.liquidation_ratio)) * 100;
+      const debtCeiling =
+        Number(parameter.debt_limit / 10 ** 6) - Number(collateral.amount);
```

Comment:
why the subtraction? debt_limit sounds like that is the debt ceiling

Reply:
i try to re-produce like a kava FE, I might be mistaken 🙏

Reply:
<img width="930" alt="image" src="https://user-images.githubusercontent.com/11234402/198633838-4241369e-125b-49a2-ae7d-2f19a034f136.png">